server/src/test/java/org/opensearch/common/breaker/ResponseLimitSettingsTests.java

```diff
@@ -0,0 +1,228 @@
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            (Metadata) null,
+            ResponseLimitSettings.LimitEntity.INDICES,
+            0
+        );
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreachedForNullRoutingTableExpectNotBreached() {
+        final Settings settings = Settings.builder().build();
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        final ResponseLimitSettings responseLimitSettings = new ResponseLimitSettings(clusterSettings, settings);
+        final boolean breached = ResponseLimitSettings.isResponseLimitBreached(
+            (RoutingTable) null,
+            ResponseLimitSettings.LimitEntity.INDICES,
+            0
+        );
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreachedForNullLimitEntityWithRoutingTableExpectException() {
+        final ClusterState clusterState = buildClusterState("test-index-1", "test-index-2", "test-index-3");
+        expectThrows(
+            IllegalArgumentException.class,
+            () -> ResponseLimitSettings.isResponseLimitBreached(clusterState.getRoutingTable(), null, 4)
+        );
+    }
+
+    public void testIsResponseLimitBreachedForNullLimitEntityWithMetadataExpectException() {
+        final ClusterState clusterState = buildClusterState("test-index-1", "test-index-2", "test-index-3");
+        expectThrows(
+            IllegalArgumentException.class,
+            () -> ResponseLimitSettings.isResponseLimitBreached(clusterState.getMetadata(), null, 4)
+        );
+    }
+
+    public void testIsResponseLimitBreachedForCatIndicesWithNoSettingPassedExpectNotBreached() {
+        // Don't set setting
+        final Settings settings = Settings.builder().build();
+        final boolean breached = isBreachedForIndices(settings);
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreachedForCatIndicesWithSettingDisabledExpectNotBreached() {
+        // Don't enable limit
+        final Settings settings = Settings.builder().put(CAT_INDICES_RESPONSE_LIMIT_SETTING.getKey(), -1).build();
+        final boolean breached = isBreachedForIndices(settings);
+        assertFalse(breached);
+    }
+
+    public void testIsResponseLimitBreachedForCatIndicesWithSettingEnabledExpectBreached() {
+        // Set limit of 1 index
+        final Settings settings = Settings.builder().put(CAT_INDICES_RESPONSE_LIMIT_SETTING.getKey(), 1).build();
+        final boolean breached = isBreachedForIndices(settings);
+        assertTrue(breached);
+    }
+
+    public void testIsResponseLimitBreachedForCatIndicesWithSettingEnabledExpectNotBreached() {
+        // Set limit of 5 indices
```

Comment:
Can you add tests for following cases:
1. Limit set to 1 index and pass one index name as a filter to the ```_cat/indices``` call.
2. Limit set to 1 index and pass two index names as a filter to the ```_cat/indices``` call.
3. Similar tests for shards and segments.

Reply:
```
public static boolean isResponseLimitBreached(final RoutingTable routingTable, final LimitEntity limitEntity, final int limit) 

public static boolean isResponseLimitBreached(final Metadata metadata, final LimitEntity limitEntity, final int limit)
```

These unit tests are around isResponseLimitBreached method which takes RoutingTable or Metadata as input that has already been filtered before we pass the values.

I have tested manually.

Reply:
> Ingest 4 indices

sumitsb@bcd074a746f6 OpenSearch % curl -XGET "http://localhost:9200/_cat/indices/"  
yellow open 1index BNLMSC3uRyiDVIM4-dEjkw 1 1 1 0 4.7kb 4.7kb
yellow open 4index UXKQ9OTbTZKjUXh5oqlDug 1 1 1 0 4.7kb 4.7kb
yellow open 3index TyqakCCZTpSM589g2Dkx2w 1 1 1 0 4.7kb 4.7kb
yellow open 2index juYqKBBkS1-kzWsTThCipA 1 1 1 0 4.7kb 4.7kb

> Update setting of indices


sumitsb@bcd074a746f6 OpenSearch % curl -X PUT "localhost:9200/_cluster/settings" -H 'Content-Type: application/json' -d'
{               
  "transient" : {
    "cat.indices.response.limit.number_of_indices" : 2 
  }
}'
{"acknowledged":true,"persistent":{},"transient":{"cat":{"indices":{"response":{"limit":{"number_of_indices":"2"}}}}}}% 

>> Cat indices works when filtered with 1 index                     
                                                              
sumitsb@bcd074a746f6 OpenSearch % curl -XGET "http://localhost:9200/_cat/indices/1ind*"                                 
yellow open 1index BNLMSC3uRyiDVIM4-dEjkw 1 1 1 0 4.8kb 4.8kb

> Cat indices works when filtered with 2 index                     

sumitsb@bcd074a746f6 OpenSearch % curl -XGET "http://localhost:9200/_cat/indices/1*,2*"
yellow open 1index BNLMSC3uRyiDVIM4-dEjkw 1 1 1 0 4.8kb 4.8kb
yellow open 2index juYqKBBkS1-kzWsTThCipA 1 1 1 0 4.8kb 4.8kb

> Cat indices fails when filtered with 3 index                     

sumitsb@bcd074a746f6 OpenSearch % curl -XGET "http://localhost:9200/_cat/indices/1*,2*,3*"
{"error":{"root_cause":[{"type":"response_limit_breached_exception","reason":"Too many indices requested.","response_limit":2,"limit_entity":"INDICES"}],"type":"response_limit_breached_exception","reason":"Too many indices requested.","response_limit":2,"limit_entity":"INDICES"},"status":429}% 

> Cat indices fails when requested without filters    

sumitsb@bcd074a746f6 OpenSearch % curl -XGET "http://localhost:9200/_cat/indices/"     
{"error":{"root_cause":[{"type":"response_limit_breached_exception","reason":"Too many indices requested.","response_limit":2,"limit_entity":"INDICES"}],"type":"response_limit_breached_exception","reason":"Too many indices requested.","response_limit":2,"limit_entity":"INDICES"},"status":429}%

Reply:
Testing for cat/shards

> Cat shards without any limit applied

curl -XGET "http://localhost:9200/_cat/shards/"         
1index 0 p STARTED    1 4.8kb 127.0.0.1 runTask-0
1index 0 r UNASSIGNED                   
4index 0 p STARTED    1 4.8kb 127.0.0.1 runTask-0
4index 0 r UNASSIGNED                   
3index 0 p STARTED    1 4.8kb 127.0.0.1 runTask-0
3index 0 r UNASSIGNED                   
2index 0 p STARTED    1 4.8kb 127.0.0.1 runTask-0
2index 0 r UNASSIGNED                 

> Apply limit

curl -X PUT "localhost:9200/_cluster/settings" -H 'Content-Type: application/json' -d'
{               
  "transient" : {
    "cat.shards.response.limit.number_of_shards" : 2  
  }
}'
{"acknowledged":true,"persistent":{},"transient":{"cat":{"shards":{"response":{"limit":{"number_of_shards":"2"}}}}}}%
                               
> Cat indices works when filtered with 1 index or 2 shards
                                                       
curl -XGET "http://localhost:9200/_cat/shards/1*"                                     
1index 0 p STARTED    1 4.8kb 127.0.0.1 runTask-0
1index 0 r UNASSIGNED             

> Cat indices fails when filtered with 2 index or 4 shards 

curl -XGET "http://localhost:9200/_cat/shards/1*,2*"
{"error":{"root_cause":[{"type":"response_limit_breached_exception","reason":"Too many shards requested.","response_limit":2,"limit_entity":"SHARDS"}],"type":"response_limit_breached_exception","reason":"Too many shards requested.","response_limit":2,"limit_entity":"SHARDS"},"status":429}%                

> Cat indices fails when requesting all

curl -XGET "http://localhost:9200/_cat/shards/"     
{"error":{"root_cause":[{"type":"response_limit_breached_exception","reason":"Too many shards requested.","response_limit":2,"limit_entity":"SHARDS"}],"type":"response_limit_breached_exception","reason":"Too many shards requested.","response_limit":2,"limit_entity":"SHARDS"},"status":429}%

Reply:
Thanks @sumitasr for the testing details.